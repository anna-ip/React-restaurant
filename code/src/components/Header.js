import React from 'react'

const Header = () => {
  return (
    <div className="header-container">
      {/* background image */}
      <h1>The Diner</h1>
    </div>
  )
}

export default Header
