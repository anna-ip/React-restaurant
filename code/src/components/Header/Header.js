import React from 'react'
import { container } from './Header.module.scss'

const Header = () => {
  return (
    <div className={container}>
      <h1>The Diner</h1>
    </div>
  )
}

export default Header
