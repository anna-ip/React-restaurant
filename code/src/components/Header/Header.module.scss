.container {
  background: green;
  border: 1px dotted black;
}
