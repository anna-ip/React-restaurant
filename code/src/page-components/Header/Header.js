import React from 'react'
import NavBar from '../../components/Nav/NavBar/NavBar'
import styles from './Header.module.scss'

const Header = () => {
  return (
    <>
      <NavBar />
      <h1 className={styles.logo}>The Diner</h1>
    </>
  )
}

export default Header
