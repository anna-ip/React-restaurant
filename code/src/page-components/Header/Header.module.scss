@import '../../styles/_variables.scss';

.logo {
  font-weight: 600;
  font-size: 3rem;
  color: $font-color;
  text-decoration: overline;
}
