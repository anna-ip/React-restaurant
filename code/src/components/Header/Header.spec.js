import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'
import { shallow } from 'enzyme'
import Header from './Header'

describe('<Header/>', () => {
  test('<Header/> renders correctly', () => {
    const tree = renderer.create(<Header />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('<Header/> has a heading with the Restaurant name', () => {
    render(<Header />)
    expect(
      screen.getByRole('heading', { name: 'The Diner' })
    ).toBeInTheDocument()
  })

  test('<Header/> contains the Restaurant name', () => {
    const wrapper = shallow(<Header />)
    const heading = wrapper.find('h1')
    const result = heading.text()
    expect(result).toBe('The Diner')
  })
})
