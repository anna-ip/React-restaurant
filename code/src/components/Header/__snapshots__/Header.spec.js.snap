// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> <Header/> renders correctly 1`] = `
<div>
  <h1
    className="logo"
  >
    The Diner
  </h1>
</div>
`;
